duckshoot.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python
+# coding: Latin
+
+import json
+from my_button import MyScale
+# Load all standard tools for image processing challenges
+from img_base_class import *
+
+
+# Image stream processing thread
+class StreamProcessor(threading.Thread):
+    def __init__(self, screen=None, camera=None, drive=None, colour="any"):
+        super(StreamProcessor, self).__init__()
+        self.camera = camera
+        image_width, image_height = self.camera.resolution
+        self.drive = drive
+        self.drive.should_normalise_motor_speed = False
+        self.screen = screen
+        self.stream = picamera.array.PiRGBArray(camera)
+        self.event = threading.Event()
+        self.terminated = False
+        self.MAX_AREA = 2000  # Largest target to shoot at
+        self.MIN_CONTOUR_AREA = 400
+        self.MAX_TARGET_SIZE = 2000
+        self.MAX_TARGET_WIDTH = 80
+        self.AIM_OFFSET = 45
+        self._colour = colour
+        self.found = False
+        self.cycle = 0
+        self.target_number = 0
+        self.menu = False
+        self.last_t_error = 0
+        self.TURN_P = 0.8
+        self.TURN_D = 0.4
+        self.CROP_WIDTH = 320
+        self.CROP_HEIGHT = 60
+        self.image_centre_x = self.CROP_WIDTH / 2.0
+        self.image_centre_y = self.CROP_HEIGHT / 2.0
+        self.CROP_H_OFFSET = 160
+        self.CROP_V_OFFSET = 180
+        self.colour_bounds = json.load(open('duckshoot.json'))
```

Comment:
this will leave the handle for 'duckshoot.json' open, as it is never explicitly closed. Unless you have a good reason for doing so, one should always interact with files and file like objects by means of a context manager. see #101 for more details.